service/service.go

```diff
@@ -0,0 +1,38 @@
+package service
+
+import (
+	"context"
+
+	"github.com/SigNoz/signoz-otel-collector/opamp"
+	"go.uber.org/zap"
+)
+
+// Service is the interface for the Opamp service
+// which manages the Opamp connection and collector
+// lifecycle
+//
+// main function will create a new service and call
+// service.Start(ctx) and service.Shutdown(ctx)
+// on SIGINT and SIGTERM
+type Service interface {
+	Start(ctx context.Context) error
+	Shutdown(ctx context.Context) error
+}
+
+type service struct {
+	l      *zap.Logger
+	client opamp.Client
+
+	managerConfigPath   string
+	collectorConfigPath string
```

Comment:
Do we need `managerConfigPath` and `collectorConfigPath` as part of the service struct?

Reply:
No, they are not needed here.